core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -2305,6 +2308,30 @@ private static class LastDayImplementor extends MethodNameImplementor {
     }
   }
 
+  /** Implementor for the {@code SAFE_MULTIPLY} function. */
+  private static class SafeArithmeticImplementor extends MethodNameImplementor {
+    SafeArithmeticImplementor() {
+      super("safeMultiply", NullPolicy.STRICT, false);
+    }
+
+    @Override Expression implementSafe(final RexToLixTranslator translator,
+        final RexCall call, final List<Expression> argValueList) {
+      Expression arg0 = toLong(argValueList.get(0), call.operands.get(0));
+      Expression arg1 = toLong(argValueList.get(1), call.operands.get(1));
+      return Expressions.call(SqlFunctions.class, "safeMultiply", arg0, arg1);
+    }
+
+    // Because BigQuery treats all int types as aliases for INT64 (Java's LONG)
+    // they can all be converted to LONG to minimize entries in the SqlFunctions class.
+    private Expression toLong(Expression arg, RexNode node) {
```

Comment:
As we only convert arg to long as when it is int type, but return the origin `arg` for other cases, it would be better to use a more proper method name to describe what this method doas.

Reply:
Good point. Changed to `convertType`, let me know if this works for you.